docs/build_stacks_rst.py

```diff
@@ -0,0 +1,124 @@
+import os
+import pathlib
+import re
+import subprocess
+
+import jinja2
+from rstcloth.rstcloth import RstCloth
+
+
+template_html = """    <div class="panel-group" id="accordion-{{ id }}">
+      <div class="panel panel-default">
+        <div class="panel-heading">
+          <h4 class="panel-title">
+            <a data-toggle="collapse" data-parent="#accordion-{{ id }}" href="#collapse1-{{ id }}">
+            conda list
+            </a>
+          </h4>
+        </div>
+        <div id="collapse1-{{ id }}" class="panel-collapse collapse">
+          <div class="panel-body">
+          <pre>{{ conda_list | indent(4) | e }}</pre>
+          </div>
+        </div>
+      </div>
+      <div class="panel panel-default">
+        <div class="panel-heading">
+          <h4 class="panel-title">
+            <a data-toggle="collapse" data-parent="#accordion-{{ id }}" href="#collapse2-{{ id }}">
+             metadata</a>
+          </h4>
+        </div>
+        <div id="collapse2-{{ id }}" class="panel-collapse collapse">
+          <div class="panel-body">
+           <table class="table table-condensed table-hover">
+            <tbody>
+           {% for key, value in metadata.items() %}
+             <tr><td>{{ key | e }}</td><td>{{ value | e }}</td></tr>
+           {% endfor %}
+            </tbody>
+           </table>
+          </div>
+        </div>
+      </div>
+    </div>
+"""
+template = jinja2.Template(template_html)
+
+
+# https://stackoverflow.com/questions/295135/turn-a-string-into-a-valid-filename
+def to_valid_filename(s):
+    s = str(s).strip().replace(' ', '_')
+    return re.sub(r'(?u)[^-\w.]', '', s).lower()
+
+def normalize_document_path(prefix_and_name, extension=''):
+    if len(prefix_and_name)==0:
+        return
+    path = os.path.join(*[to_valid_filename(p) for p in prefix_and_name])
+    return path + extension
+
+
+class StacksRSTBuilder:
+
+    def __init__(self, images, output_dir, actually_load=True):
+        self.images = images
+        self.output_dir = output_dir
+        self.actually_load = actually_load
+
+    def conda_list(self, image):
+        return "docker run ... conda list"
+        # args = [
+        #     'docker', 'run', '-it',
+        #     '--entrypoint=""',
+        #     image, 
+        #     'conda', 'list']
+        # joined_args = " ".join(args)
+        # print('running ', joined_args)
+        # out = subprocess.check_output(joined_args, shell=True)
+        # out = out.decode().replace('#', '') + '\n'
+        # print(out, type(out))
+        # return out
```

Comment:
This isn't working yet. We need to figure out the best way to clean up the output from `conda-list` so it can be rendered in rst/html.